src/ripple/basics/impl/PerfLogImp.h

```diff
@@ -0,0 +1,250 @@
+            std::uint64_t duration {0};
+            std::string label;
+            std::unique_ptr<Json::StaticString> labelSs;
+            mutable std::mutex mut;
+
+            Rpc() = default;
+
+            Rpc(Rpc const& orig)
+                : started (orig.started)
+                , finished (orig.finished)
+                , duration (orig.duration)
+                , label (orig.label)
+            {
+                if (orig.labelSs)
+                {
+                    labelSs = std::make_unique<Json::StaticString>(
+                        orig.labelSs->c_str());
+                }
+            }
+        };
+
+        /**
+         * Job Queue task performance counters.
+         */
+        struct Jq
+        {
+            std::uint64_t queued {0};
+            std::uint64_t started {0};
+            std::uint64_t finished {0};
+            std::uint64_t queuedDuration {0};
+            std::uint64_t runningDuration {0};
+            std::string label;
+            std::unique_ptr<Json::StaticString> labelSs;
+            mutable std::mutex mut;
+
+            Jq() = default;
+
+            Jq(Jq const& orig)
+            : queued (orig.queued)
+            , started (orig.started)
+            , finished (orig.finished)
+            , queuedDuration (orig.queuedDuration)
+            , runningDuration (orig.runningDuration)
+            , label (orig.label)
+            {
+                if (orig.labelSs)
+                {
+                    labelSs = std::make_unique<Json::StaticString>(
+                        orig.labelSs->c_str());
+                }
+            }
+        };
+
+        std::unordered_map<std::string, Rpc> rpc_;
+        std::unordered_map<std::underlying_type_t<JobType>, Jq> jq_;
+        std::vector<std::pair<std::underlying_type_t<JobType>,
+            std::chrono::steady_clock::time_point>> jobs_;
+        int workers_ {0};
+        mutable std::mutex jobsMutex_;
+        std::unordered_map<std::uint64_t, std::pair<std::string,
+            std::chrono::steady_clock::time_point>> methods_;
+        mutable std::mutex methodsMutex_;
+
+        /**
+         * Initialize map of RPC method counters
+         *
+         * @param labels method names
+         */
+        void populateRpc(std::vector<char const*> const& labels);
+
+        /**
+         * Initialize map of Job Queue task counters
+         *
+         * @param jobs Job Queue types and text labels
+         */
+        void populateJq(std::unordered_map<std::underlying_type_t<JobType>,
+            std::string> const& jobs);
+        Json::Value countersJson() const;
+        Json::Value currentJson() const;
+    };
+
+    Setup const setup_;
+    Counters counters_;
+    std::ofstream logFile_;
+    std::thread thread_;
+    std::mutex mutex_;
+    std::condition_variable cond_;
+    std::chrono::time_point<std::chrono::system_clock> lastLog_;
+    std::string const cHostname_ {beast::getComputerName()};
```

Comment:
Unless this value is **only** ever displayed to admin users, then prefer using `NetworkOPs::getHostId` which masks the hostname to avoid potentially leaking sensitive information like a hostname.

Reply:
It is only logged to the file, on the host itself.